Integrate eslint errors into msbuild (#11)
Lombiq/Gulp-Extensions: Readme.md

const jsTargets = require('path/to/Lombiq.Gulp.Extensions/Tasks/js-targets');

const path = './Assets/Scripts/'
const destination = './directory-to-copy-into'

gulp.task('build:js', () => jsTargets.compile(path, destination));
```

The rules are found in 3 files:

> - *eslint-base.js* these rules come from the [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript).
> - *lombiq-base.js* these rules are Lombiq overrides for the Airbnb rules.
> - *.eslintrc* in this file you can define your own overriding rules

Reviewer: ```suggestion
- *eslint-base.js*: These rules come from the [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript).
- *lombiq-base.js*: These rules are Lombiq overrides for the Airbnb rules.
- *.eslintrc*: In this file you can define your own overriding rules.
```

Since these files will be copied int your solution's root you may want to gitignore it:
```
/src/.eslintrc
/src/eslint-base.js
/src/lombiq-base.js

> ```

Reviewer: Lásd a copy-zásról a többi commentemet.

Rules can be found in the [ESLint documentation](https://eslint.org/docs/rules/).

> The build output or task runner will show you a all the ESLint rule violations in a detailed manner.

Reviewer: ```suggestion
The build output or task runner will show you all of the ESLint rule violations in a detailed manner.
```

> If you want to integrate ESLint into MSBuild you need to include Lombiq's [NPM-Targets](https://github.com/Lombiq/NPM-Targets) too.

Reviewer: ```suggestion
If you want to integrate ESLint into MSBuild you need to include Lombiq's [NPM MSBuild Targets](https://github.com/Lombiq/NPM-Targets) too.
```

Reviewer: Az ilyen suggested change-eket amúgy egyből commitolhatod innen a PR alól.

In the project ESLint needs use, you need to import these files in the `.csproj` file:
```
<Import Project="path\to\Lombiq.Npm.Targets\Lombiq.Npm.Targets.props" />
<Import Project="path\to\Lombiq.Npm.Targets\Lombiq.Npm.Targets.targets" />
<Import Project="path\to\Lombiq.Gulp.Extensions\Lombiq.Gulp.Extensions.targets"/>
```

> Then a warning will be sent to the error list if ESLint finds a rule violation.

Reviewer: ```suggestion
Then a warning will be sent to the MSBuild output ESLint finds a rule violation.
```

Author: Az error list-be küldi ez a része ugye, a Build outputos része az a 89. sor

Reviewer: Aha, valóban.

> If you want to use [Visual Studio's built in ESLint](https://docs.microsoft.com/en-us/visualstudio/ide/reference/options-text-editor-javascript-linting?view=vs-2019) aswell, after the first build of Gulp Extensions, the built in ESLint will use the rules found in this extension.

Reviewer: ```suggestion
If you want to use [Visual Studio's built-in ESLint](https://docs.microsoft.com/en-us/visualstudio/ide/reference/options-text-editor-javascript-linting?view=vs-2019) as well, after the first build of Gulp Extensions, the built-in ESLint will use the rules found in this extension.
```

Ez nem egészen értem, hogy mit jelent. Azt akarod mondani, hogy VS-ben is látszani fognak az editorban is a warningok? Mert akkor még tedd hozzá.

const path = './Assets/Scripts/'

> const destination = './directory-to-cop-into'

Reviewer: ```suggestion
const destination = './directory-to-copy-into'
```

gulp.task('build:js', () => jsTargets.compile(path, destination));
```

> The build output or task runner will show you a list of all the ESLint rule violations.

Reviewer: Meg a VS Error List window, meg `dotnet build`. Nézd át a .NET Analyzers guide-jait, az ottanihoz hasonlóan, még ha nem is pont ugyanúgy, megfelelő részletességgel, kellene itt is leírni, hogy hogyan kell ezeket használni: https://github.com/Lombiq/.NET-Analyzers#guides.

> The base rules for the linter are found in the *eslint-base.js* and the *lombiq-base.js* files, you can add your own rules in the `.eslintrc` file overriding these base rules.

Reviewer: Hogyan lehet override-olni az .eslintrc-ben? Ugye a projectet nem akarjuk minden solutionhöz forkolni, ha viszont ott változtatunk rajta, akkor minden buildnél felülírja innen.
